File: apps/api/src/lib/serializers/flight.ts
import type { Flight, User } from "../../db/schema";

import type { Serializer } from ".";

export const FlightSerializer: Serializer<Flight> = {
  item: (item: Flight, attrs: Record<string, any>, currentUser?: User) => {
    return {
      id: item.publicId,
      name: item.name,
      description: item.description,
      public: item.public,
      createdAt: item.createdAt,
    };
  },
};

File: src/server/db/model/flight.go
package model

import (
	"time"
)

type Flight struct {
	ID          uint64 `gorm:"primaryKey" json:"id"`
	PublicID    string `json:"public_id"`
	Name        string `json:"name"`
	Description string `json:"description"`
	Public      bool   `json:"public"`

	CreatedByID uint64    `json:"created_by_id"`
	CreatedAt   time.Time `json:"created_at"`

	CreatedBy User
}

func (Flight) TableName() string {
	return "flight"
}

type Flights []*Flight
